pkg/controller/storagecluster/external_resources.go

```diff
@@ -271,3 +294,28 @@ func (r *ReconcileStorageCluster) createExternalStorageClusterSecret(sec *corev1
 	}
 	return nil
 }
+
+// To check if endpoint is a VALID ip and is REACHABLE or not
+func validateMonitoringEndpoint(monitoringIP string, monitoringPort string, reqLogger logr.Logger) error {
+	_, err := net.LookupIP(monitoringIP)
+	if err != nil {
+		reqLogger.Error(err, "Monitoring endpoint is not a valid IPv4 Ip.")
+		return err
+	}
+	endpoint := monitoringIP + ":" + monitoringPort
+	err = checkEndpointReachable(endpoint, 5*time.Second, reqLogger)
+	if err != nil {
+		reqLogger.Error(err, fmt.Sprintf("Monitoring Endpoint (%s) is not reachable.", endpoint))
+		return err
+	}
+	return nil
+}
+
+func checkEndpointReachable(endpoint string, timeout time.Duration, reqLogger logr.Logger) error {
```

Comment:
Is this necessary?

I think we can skip this test. If IP is valid, pass it and create service and servicemonitor. If it is not reachable Prometheus will mark the target accordingly. We can probably build an alert about it then?

The problem with this approach is it's not very reliable (IMO). A timeout of 5 sec and a oneof operation doesn't reflect the real state of connection. What if it becomes unreachable after 5s? or it's unreachable for 10s and is reachable after that?

Reply:
This for the validation as to when OCP is not able to reach ceph-mgr cluster, even if the IP is valid. We stop the control flow here. Thus Cephcluster and other objects are not created on the first hand. User can then just delete the storage cluster quickly, check the connectivity and then again try
If later on the connection goes bad its an infrastructure problem and we go the Prometheus way. What do you say ?@umangachapagain

Reply:
Isn't this done when CephCluster is getting Connected via Rook? 
If CephCluster is Connected we know OCP can reach it. Do we need same validations on other layers?

@leseb any thoughts on this?

Reply:
This check seems a bit redundant to what Rook does internally and somehow "weak" as well since it's just a dial on the address **but** I like the fact we fail early so I'd keep it around. Also, 5s is more than enough, 5s network-wise feels like an eternity, so if after 5s we don't get any reply it's fair to assume the endpoint is not reachable.